### ISSUE TITLE
get_architecture_value and ndk_platform_triple is no longer used in test/lit.cfg and can be removed

### ISSUE THREAD

**Issue by WeZZard** [swiftlang/swift#63907]
**Description**
I configured pylance for lit config scripts in the `test` directory and find that in current main branch, at the line 1572 of the file test/lit.cfg there is:

```python
            result = kwards["aarch64"]
```

which should be:

```python
            result = kwargs["aarch64"]
```

which make the variable `ndk_platform_triple` below receives incorrect value.

The variable `ndk_platform_triple` is introduced in the pull request [[android] Modify test scripts for aarch64 and modern NDKs.](https://github.com/apple/swift/pull/19503/files#diff-a041dc49aeef2cbc59d598b3df453d06221ec2704c2c076590d5835cbb77de44R940) but the use sites of the variable `ndk_platform_triple` are removed in the pull request [[android] Modify test scripts for aarch64 and modern NDKs.](https://github.com/apple/swift/pull/34491/files) and [[android] Update to NDK 23b](https://github.com/apple/swift/pull/39921). Thus the `get_architecture_value` function and `ndk_platform_triple` could be removed now.

**Environment**
swift main branch, test/lit.cfg

**Comment by finagolfin** [comment]
You are correct. The issue is that this config has not been used in awhile, but the new community Android CI uses it again, swiftlang/swift-community-hosted-continuous-integration#55. We'll look at fixing this alongside #78350, let us know if we forget.
